On Windows, run PKG_CONFIG_PATH through cygpath

MisterDA opened the pull request:
When using OCaml for Windows, `PKG_CONFIG_PATH` is usually in
Windows style. GNU Make with the `shell` function will call Cygwin's
Unix shell which doesn't recognize Windows paths. If we're on a
Windows system, and if we find `cygpath`, convert the path list to Unix
style.

dinosaure commented:
I would like to accept this PR but it seems that the AppVeyor CI (which tests `checkseum` on Windows) does not work. I'm not sure how to fix that correctly. May be `ocaml/setup-ocaml` has a Windows support and we should move on this system instead of great old AppVeyor CI.

MisterDA commented:
I've added GHA support and removed the AppVeyor scripts.
The build [succeeds on my fork](https://github.com/MisterDA/checkseum/actions/runs/1648944924).

dinosaure commented:
Thanks!
